dev-tools/omdb/src/bin/omdb/nexus.rs

```diff
@@ -3871,19 +3879,88 @@ async fn write_stream_to_sink(
     Ok(())
 }
 
+// Downloads a portion of a support bundle using range requests.
+//
+// "range" is in bytes, and is inclusive on both sides.
+async fn support_bundle_download_range(
+    client: &nexus_client::Client,
+    id: SupportBundleUuid,
+    range: (u64, u64),
+) -> anyhow::Result<impl futures::Stream<Item = anyhow::Result<bytes::Bytes>>> {
+    let range = format!("bytes={}-{}", range.0, range.1);
+    Ok(client
+        .support_bundle_download(id.as_untyped_uuid(), Some(&range))
+        .await
+        .with_context(|| format!("downloading support bundle {}", id))?
+        .into_inner_stream()
+        .map(|r| r.map_err(|err| anyhow::anyhow!(err))))
+}
+
+// Downloads all ranges of a support bundle, and combines them into a single
+// stream.
+//
+// Starts the download at "start" bytes (inclusive) and continues up to "end"
+// bytes (exclusive).
+fn support_bundle_download_ranges(
+    client: &nexus_client::Client,
+    id: SupportBundleUuid,
+    start: u64,
+    end: u64,
+) -> impl futures::Stream<Item = anyhow::Result<bytes::Bytes>> + use<'_> {
+    // Arbitrary chunk size of 100 MiB.
+    //
+    // Note that we'll still stream data in packets which are smaller than this,
+    // but we won't keep a single connection to Nexus open for longer than a 100
+    // MiB download.
+    const CHUNK_SIZE: u64 = 100 * (1 << 20);
+    futures::stream::try_unfold(
+        (start, start + CHUNK_SIZE - 1),
+        move |range| async move {
+            if end <= range.0 {
+                return Ok(None);
+            }
+
+            let stream =
+                support_bundle_download_range(client, id, range).await?;
+            let next_range = (range.0 + CHUNK_SIZE, range.1 + CHUNK_SIZE);
+            Ok::<_, anyhow::Error>(Some((stream, next_range)))
+        },
+    )
+    .try_flatten()
+}
+
 /// Runs `omdb nexus support-bundles download`
 async fn cmd_nexus_support_bundles_download(
     client: &nexus_client::Client,
     args: &SupportBundleDownloadArgs,
 ) -> Result<(), anyhow::Error> {
-    let stream = client
-        .support_bundle_download(args.id.as_untyped_uuid())
-        .await
-        .with_context(|| format!("downloading support bundle {}", args.id))?
-        .into_inner_stream();
+    let total_length = client
+        .support_bundle_head(args.id.as_untyped_uuid())
+        .await?
+        .content_length()
+        .ok_or_else(|| anyhow::anyhow!("No content length"))?;
+
+    let start = if let Some(output) = &args.output {
+        if output.exists() && args.resume {
+            output.metadata()?.len()
+        } else {
+            0
+        }
+    } else {
+        0
+    };
```

Comment:
would something like this work here:

```rust
match &args.output {
    Some(output) if output.exists() && args.resume => output.metadata()?.len(),
    _ => 0,
}
```

"Thanks, I hate it" is a fine response ;-)

Reply:
Oh, that's better! I was trying to use "if let Some(...) = ... &&  ... { }" which is unstable, but the match works